README.md

```diff
@@ -1 +1,308 @@
+    "role": "client"
+}
+```
+
+
+6. Create Client Profile
+Description: Creates a new client profile.
+Endpoint: /clients/
+Method: POST
+Request Body:
+```json
+{
+  "address": "",
+  "contactNumber": ""
+}
+```
+
+7. Get Client Profile
+Description: Retrieves the profile of the authenticated client.
+Endpoint: /clients/me
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+8. Update Client Profile
+Description: Updates the profile of the authenticated client.
+Endpoint: /clients/me
+Method: PUT
+Request Body:
+```json
+{
+  "address": ""
+}
+```
+
+9. Retrieve Chef information
+Description: Retrieves the profile of a chef Chef information based on their ID.
+Endpoint: /chef/me
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve.
+Request Body:
+```json
+{
+
+}
+```
+
+10. Edit Chef Profile
+Description: Allows the authenticated chef to edit their profile information.
+Endpoint: /chef/me
+Method: PUT
+Request Body:
+```json
+{
+  "restaurant": ""
+}
+```
+
+11. Retrieve Chefs by a client
+Description: Retrieves a list of all chefs by a client.
+Endpoint: /chefs
+Method: GET
+Request Body:
+```json
+{
+ 
+}
+```
+
+12. Retrieve Chef Dishes by chef
+Description: Retrieves a list of dishes offered by a certain chef.
+Endpoint: /chefs/:id/dishes
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve dishes from.
+Request Body:
+```json
+{
+  
+}
+```
+
+13. Retrieve Client's Favorite Dishes
+Description: Retrieves a list of dishes favorited by the authenticated client.
+Endpoint: /clients/me/favorite-dishes
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+14. Add Dish to Client's Favorite Dishes
+Description: Adds a dish to the list of favorite dishes for the authenticated client.
+Endpoint: /clients/me/favorite-dishes
+Method: POST
+Request Body:
+```json
+{
+  "dishId": ""
+}
+```
+
+15. Retrieve Client's Order History
+Description: Retrieves the order history of the authenticated client.
+Endpoint: /clients/me/order-history
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+16. Delete Dish from Client's Favorite Dishes
+Description: Deletes a dish from the list of favorite dishes for the authenticated client.
+Endpoint: /clients/me/favorite-dishes/:id
+Method: DELETE
+Parameters:
+id (required): The ID of the dish to remove from favorites.
+Request Params:
+```json
+{
+  "dishId": ""
+}
+```
+
+17. Create Order
+Description: Creates a new order for the authenticated client.
+Endpoint: /orders
+Method: POST
+Request Body:
+```json
+{
+  
+}
+```
+18. Create Chef Profile
+Description: Allows a user to create a chef profile to showcase their culinary offerings.
+Endpoint: /chef/
+Method: POST
+Request Body:
+```json
+{
+  {
+    "restaurant": "",
+    "location": "",
+    "openingHours": "",
+    "closingHours": "",
+    "contactNumber": "",
+    "description": ""
+}
+}
+```
+
+
+## Examples
+
+Here are some examples of how you can interact with the backend API using various endpoints:
```

Comment:
There is only 1 example, i think we should add more for chef, client and order endpoints